packages/compass-e2e-tests/helpers/commands/run-find.ts

```diff
@@ -6,10 +6,19 @@ export async function runFind(
   tabName: string,
   waitForResult: boolean
 ): Promise<void> {
-  // look up the current resultId
+  // look up the current resultId and applyId
   const initialResultId = await browser.getQueryId(tabName);
+  const initialApplyId = await browser.getApplyId(tabName);
 
-  await browser.clickVisible(Selectors.queryBarApplyFilterButton(tabName));
+  // In some rare cases in ci (for example when clicking button in a query bar
+  // right after closing a modal, clicks are not registering correctly on
+  // certain machine types, in this case we continue clicking until we see that
+  // applyId changed
+  await browser.waitUntil(async () => {
+    await browser.clickVisible(Selectors.queryBarApplyFilterButton(tabName));
+    await browser.pause(50);
+    return (await browser.getApplyId(tabName)) !== initialApplyId;
+  });
```

Comment:
So two failing tests were "should fail with time out when `maxTimeMS` set via preferences modal" and "should clone and delete document", in both cases the broken find was happening right after modal would be closed, in first case it's applying setting, in second applying "cloning" the document. Hmmm, so probably the cleanest way to address this would be to wait for modals to dissapear, but I'm also not sure I'll be able to track all these case, so maybe I'll add wait, but keep this too for now?

Reply:
And `await  insertDialog.waitForDisplayed({ reverse: true });`

Weird that we didn't already have it.